### ISSUE TITLE
Release an opensuse package

### ISSUE THREAD

**Issue by ghost** [jeromerobert/pdfarranger#151]
Would it be possible to release pdf- arranger for opensuse?

**Comment by dreua** [comment]
Hello @tabascosw and welcome to Github! 

TL;DR: Of course it's possible but you'll need to find a package maintainer on the openSUSE side.

Linux distributions (e.g. openSUSE, Fedora, Arch...) usually have a group of packagers who take care of creating and maintaining software packages for their distribution. This Github repository provides the source code that can be packaged for any distribution but I'm not sure if there is someone here with the knowledge or time to take care of an openSUSE version of pdfarranger. Rules and procedures differ between the distributions and you need at least a virtual machine with the target distribution to test that the package actually works, therefore I won't maintain packages for distributions that I don't use myself. I might do it for money, but it's probably better if you found someone who is already experienced in packaging for openSUSE. The [openSUSE forums or mailing list](https://en.opensuse.org/Portal:Support#Interactive) might be a place to ask.

Since I am not only helping with the code here but also maintaining the package on Fedora, feel free to refer to me if any problems come up with the packaging. (Fedora and suse are both RPM based.)

**Comment by dreua** [comment]
Huuh, now I feel ghosted :ghost: . I wrote my comment with the best intentions in mind and I'm sorry if it was too much or wrong in any way. Feel free to [email me](mailto:dreua@posteo.de) if there is anything about this you want to discuss.

Actually I came up with one thing we could try: I managed to build pdfarranger for openSUSE on copr (Fedora's user build service) using the [spec from rpmsphere](https://github.com/rpmsphere/source/blob/master/p/pdfarranger-1.1-1.src.rpm) which I updated to version 1.4.1. (My Fedora spec failed for some reason. Probably because it uses a makro which openSUSE doesn't know.)

I have no clue whether that package will install or work but it's probably worth a try. Let me know how it goes: https://copr.fedorainfracloud.org/coprs/dreua/pdfarranger-openSUSE/

Another way would be to use a flatpack but we don't have that yet: #37

**Comment by tabascosw** [comment]
Hello dreua, your package for Opensuse Tumbleweed did not work, it has been build for python 3.7 but Tumbleweed already uses python 3.8. So I rebuild your srpm with my system and that worked just fine, the package installed without any dependency issue but would not run, I was missing pikepdf and typelib-1_0-Poppler-0_18, after installing both packages PDFArranger is now running happily on my system.
Big thanks for that.
BTW: I am not a big fan of flatpak, too much software/libraries are installed even for the smallest software package.

**Comment by dreua** [comment]
After a few emails back and forth, we managed to fix the Copr build. I updated the [description there](https://copr.fedorainfracloud.org/coprs/dreua/pdfarranger-openSUSE/) and the [wiki page for binaries](https://github.com/jeromerobert/pdfarranger/wiki/Binary-packages)

**Comment by jeromerobert** [comment]
Thanks !

**Comment by dreua** [comment]
Update: The Copr is discontinued as someone built it on the openSUSE build service: [Office / pdfarranger](https://build.opensuse.org/package/show/Office/pdfarranger)
